procrustes/generic.py

```diff
@@ -0,0 +1,113 @@
+    remove_zero_col : bool, optional
+        If True, the zero columns on the right side will be removed.
+        Default=True.
+    remove_zero_row : bool, optional
+        If True, the zero rows on the top will be removed.
+        Default=True.
+    pad_mode : str, optional
+        Specifying how to pad the arrays, listed below. Default="row-col".
+
+            - "row"
+                The array with fewer rows is padded with zero rows so that both have the same
+                number of rows.
+            - "col"
+                The array with fewer columns is padded with zero columns so that both have the
+                same number of columns.
+            - "row-col"
+                The array with fewer rows is padded with zero rows, and the array with fewer
+                columns is padded with zero columns, so that both have the same dimensions.
+                This does not necessarily result in square arrays.
+            - "square"
+                The arrays are padded with zero rows and zero columns so that they are both
+                squared arrays. The dimension of square array is specified based on the highest
+                dimension, i.e. :math:`\text{max}(n_a, m_a, n_b, m_b)`.
+    translate : bool, optional
+        If True, both arrays are translated to be centered at origin.
+        Default=False.
+    scale : bool, optional
+        If True, both arrays are column normalized to unity. Default=False.
+    check_finite : bool, optional
+        If true, convert the input to an array, checking for NaNs or Infs.
+        Default=True.
+
+    Returns
+    -------
+    new_a : ndarray
+        The transformed ndarray array_a.
+    new_b : ndarray
+        The transformed ndarray array_b.
+    array_x : ndarray
+        The optimum symmetric transformation array.
+    e_opt : float
+        One-sided Procrustes error.
+
+    Notes
+    -----
+    Given a source matrix :math:`\mathbf{A} \in \mathbb{R}^{m \times n}` and a target matrix
+    :math:`\mathbf{B} \in \mathbb{R}^{m \times n}`, find the transformation matrix
+    :math:`\mathbf{X} \in \mathbb{R}^{n \times n}` for which :math:`\mathbf{AX}` is as close as
+    possible to :math:`\mathbf{B}`. I.e.,
+
+    .. math::
+       \text{min} \quad \|\mathbf{A} \mathbf{X} - \mathbf{B}\|_{F}^2
+
+    The optimal transformation matrix :math:`\mathbf{X}` is given by
+
+    .. math::
+        \mathbf{X} = {(\mathbf{A}^{\top}\mathbf{A})}^{-1} \mathbf{A}^{\top} \mathbf{B}
+
+    References
+    ----------
+    1. Gower, J. C. Procrustes Methods. Wiley Interdisciplinary Reviews: Computational Statistics,
+       2(4), 503-508, 2010.
+    """
+    # check inputs
+    new_a, new_b = setup_input_arrays(array_a, array_b, remove_zero_col, remove_zero_row,
+                                      pad_mode, translate, scale, check_finite)
+    # compute the generic solution
+    a_inv = np.linalg.inv(np.dot(new_a.T, new_a))
```

Comment:
I think it's safer to use generalized inverse; see: https://numpy.org/doc/stable/reference/generated/numpy.linalg.pinv.html

Reply:
Fixed in 904f22b. Thanks for pointing it out.